src/controlplane/client/connector.go

```diff
@@ -98,133 +98,3 @@ func (dp *defaultConnector) probeEndpoint(url string, client *http.Client) error
 
 	return nil
 }
-
-func (dp *defaultConnector) newHTTPClient(endpoint config.Endpoint) (*http.Client, error) {
-	client := &http.Client{Timeout: DefaultTimout}
-
-	// Here we're using the default http.Transport configuration, but with a modified TLS config.
-	// For some reason the DefaultTransport is casted to an http.RoundTripper interface, so we need to convert it back.
-	t := http.DefaultTransport.(*http.Transport).Clone()
-	t.TLSClientConfig = &tls.Config{InsecureSkipVerify: endpoint.InsecureSkipVerify}
-	client.Transport = t
-
-	if err := dp.configureAuthentication(client, endpoint); err != nil {
-		return nil, fmt.Errorf("configuring auth: %w", err)
-	}
-
-	return client, nil
-}
-
-func (dp *defaultConnector) configureAuthentication(httpClient *http.Client, endpoint config.Endpoint) error {
-	if endpoint.Auth == nil {
-		dp.logger.Debugf("No authentication configured for %q, connection will be attempted anonymously", endpoint.URL)
-
-		return nil
-	}
-
-	if strings.EqualFold(endpoint.Auth.Type, bearerAuth) {
-		dp.logger.Debugf("Using kubernetes token to authenticate request to %q", endpoint.URL)
-
-		httpClient.Transport = transport.NewBearerAuthRoundTripper(dp.inClusterConfig.BearerToken, httpClient.Transport)
-
-		return nil
-	}
-
-	if strings.EqualFold(endpoint.Auth.Type, mTLSAuth) {
-		dp.logger.Debugf("Using mTLS to authenticate request to %q", endpoint.URL)
-
-		tlsConfig, err := dp.getTLSConfigFromSecret(endpoint.Auth.MTLS)
-		if err != nil {
-			return fmt.Errorf("could not load TLS configuration: %w", err)
-		}
-
-		httpClient.Transport = &http.Transport{
-			TLSClientConfig: tlsConfig,
-		}
-
-		return nil
-	}
-
-	return fmt.Errorf("unknown authorization type %q", endpoint.Auth.Type)
-}
-
-func (dp *defaultConnector) getTLSConfigFromSecret(mTLSConfig *config.MTLS) (*tls.Config, error) {
-	if mTLSConfig == nil {
-		return nil, fmt.Errorf("mTLS config cannot be nil")
-	}
-
-	if mTLSConfig.TLSSecretName == "" {
-		return nil, fmt.Errorf("mTLS secret name cannot be empty")
-	}
-
-	namespace := mTLSConfig.TLSSecretNamespace
-	secretName := mTLSConfig.TLSSecretName
-	if namespace == "" {
-		dp.logger.Debugf("TLS Secret name configured, but not TLS Secret namespace. Defaulting to `default` namespace.")
-		namespace = "default"
-	}
-
-	secret, err := dp.kc.CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("could not find secret %q containing TLS configuration: %w", secretName, err)
-	}
-
-	var cert, key, cacert []byte
-
-	var ok bool
-	if cert, ok = secret.Data["cert"]; !ok {
-		return nil, fmt.Errorf("could not find TLS certificate in `cert` field in secret %q", secretName)
-	}
-
-	if key, ok = secret.Data["key"]; !ok {
-		return nil, fmt.Errorf("could not find TLS key in `key` field in secret %q", secretName)
-	}
-
-	cacert, hasCACert := secret.Data["cacert"]
-	insecureSkipVerifyRaw, hasInsecureSkipVerify := secret.Data["insecureSkipVerify"]
```

Comment:
the posibility to configure the ca verification has been removed from the secrets, now it needs to be configured in the endpoint configuration (as the others auth methods)